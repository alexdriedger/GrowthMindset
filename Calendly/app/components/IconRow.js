import React, { Component } from 'react';
import { TextInput, View, StyleSheet } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 16,
  },
  textBox: {
    flex: 1,
    height: 50,
    padding: 8,
    paddingLeft: 40,
    fontSize: 22,
  },
});

class IconRow extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Icon name={this.props.icon} size={30} color="black" />
        <TextInput
          style={styles.textBox}
          onChangeText={this.props.onChange}
          placeholder={this.props.defaultText}
          value={this.props.text}
        />
      </View>
    );
  }
}

IconRow.propTypes = {
  icon: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
  defaultText: PropTypes.string.isRequired,
};

export default IconRow;
