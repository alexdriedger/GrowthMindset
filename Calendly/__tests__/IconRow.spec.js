import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import IconRow from '../app/components/IconRow';

it('renders correctly', () => {
  const tree = renderer
    .create(<IconRow icon="map" onChange={() => {}} text="Testing" defaultText="T" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
