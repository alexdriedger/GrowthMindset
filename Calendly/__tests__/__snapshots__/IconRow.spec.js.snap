// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "padding": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
          "fontSize": 30,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î•›
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="T"
    style={
      Object {
        "flex": 1,
        "fontSize": 22,
        "height": 50,
        "padding": 8,
      }
    }
    value="Testing"
  />
</View>
`;
